CHANGELOG.md

```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `mypy` for search space
 - Class hierarchy for objectives
 - `mypy` for objectives
+- `Kernel` base class allowing to specify kernels
```

Comment:
@AdrianSosic @Scienfitz all changes included :) Two things are still kind of open, I have kept two threads discussing those open since there were intersections/comments out of date and so on:
1. The tests: I have removed the strategy for creating all of the `Fraction` objects and instead just included a test with the three correponding fraction objects.
2. The function for getting the stuff from init: I have renamed it to `get_init_arguments` since this is what it actually does and adjusted variable names and docstring. Since this will also be used in other contexts, I want to make sure that the naming I chose is correct and makes sense.

Also, the _convert_nu function underwent some major change, but since this was discussed in our internal chat, I did not keep a thread on this - just fyi. Once you have giving your "gucci" regarding the content, I will clean the commit history, force-push and merge.

Reply:
@Scienfitz @AdrianSosic final rebase and git history cleaning done, the PR is now also down to a reasonable number of commits :D Please given your final approve/comments and feel free to ping me for merge or merge yourselves

Reply:
Gorgeous 🐕 🥇 doge much approve. Ready for meeeerge 🚀